File: ts/lcof/06.ts
import { ListNode } from '../class'

function reversePrint(head: ListNode | null): number[] {
  const stack: number[] = [], ans: number[] = []
  let cur = head
  while (cur) {
    stack.push(cur.val)
    cur = cur.next
  }
  while(stack.length) {
    ans.push(stack.pop())
  }

  return ans
};

File: py3/lcof/06.py
from typing import List

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        res = []
        ptr = head
        while ptr:
            res.append(ptr.val)
            ptr = ptr.next
        res.reverse()
        return res